pages/user/[username]/index.tsx

```diff
@@ -103,10 +103,10 @@ export async function handleUserSSR({ params }: GetServerSidePropsContext<{ user
   });
 
   const socialCardUrl = `${String(process.env.NEXT_PUBLIC_OPENGRAPH_URL ?? "")}/users/${username}`;
-  let ogImage = socialCardUrl;
-  const ogReq = await fetch(`${socialCardUrl}/metadata`); //statsu returned: 204 or 304 or 404
-  if(ogReq.status === 204 || ogReq.status === 304) {
-    ogImage = ogReq.headers.get("x-amz-meta-location") ?? socialCardUrl;
+  const ogReq = await fetch(`${socialCardUrl}/metadata`); //status returned: 204 or 304 or 404
```

Comment:
```
let ogImage = socialCardUrl;
axios.get(`${socialCardUrl}/metadata`, {
      withCredentials: false,
      validateStatus: (status) => [204, 304, 404].includes(status)
    })
      .then(({ headers }) => {
        ogImage = headers.get("x-amz-meta-location");
        return axios.head(`${socialCardUrl}`, {
          withCredentials: false
        });
      });
```

and move the whole block after `const user = await req.json() as DbUser;` as the username would be validated in the database call

Reply:
@0-vortex We don't have axios in the project. Do you want me to specifically install it or just do the equivalent with `fetch`?

Reply:
Just use fetch

Reply:
idea is you don't get the content back and don't need to await it, just generate the image with `method: "HEAD"`, linked axios code because it exists in typesync package and I don't know fetch equivalents 😅